crates/primitives/starknet/src/transaction/mod.rs

```diff
@@ -795,7 +798,7 @@ impl Transaction {
         AccountTransactionContext {
             transaction_hash: tx.transaction_hash(),
             max_fee: tx.max_fee(),
-            version: tx.version(),
+            version: calculate_transaction_version(self.is_query, tx.version()),
```

Comment:
just `tx.version` no ?

Reply:
Do you mean why is it a function and not just `tx.version`? `DeclareTransaction` doesn't have a `version` field as well and instead has a function which is why we use that. I am not sure if this was the question